src/SupabaseClient.ts

```diff
@@ -319,14 +321,13 @@ export default class SupabaseClient<
     })
   }
 
-  private _listenForAuthEvents() {
-    let data = this.auth.onAuthStateChange((event, session) => {
+  private async _listenForAuthEvents() {
```

Comment:
the caller doesn't use an `await`, isn't this an issue?

Reply:
yes indeed, changing